src/TokenRatesController.ts

```diff
@@ -158,19 +159,19 @@ export class TokenRatesController extends BaseController<TokenRatesConfig, Token
 	 * @returns Promise resolving when this operation completes
 	 */
 	async updateExchangeRates() {
-		if (this.tokenList.length === 0) {
+		if (this.disabled || this.tokenList.length === 0) {
 			return;
 		}
 		const newContractExchangeRates: { [address: string]: number } = {};
 		const { nativeCurrency } = this.config;
 		const pairs = this.tokenList.map((token) => `pairs[]=${token.address}/${nativeCurrency}`);
 		const query = pairs.join('&');
-		const { prices = [] } = await this.fetchExchangeRate(query);
+		const { prices } = (await this.fetchExchangeRate(query)) || /* istanbul ignore next */ { prices: [] };
 		prices.forEach(({ pair, price }) => {
 			const address = toChecksumAddress(pair.split('/')[0].toLowerCase());
 			newContractExchangeRates[address] = typeof price === 'number' ? price : 0;
 		});
-		this.update({ contractExchangeRates: newContractExchangeRates });
+		this.update({ contractExchangeRates: { ...newContractExchangeRates } });
```

Comment:
newContractExchangeRates is a new object so there's no need to use the spread operator to create a new one